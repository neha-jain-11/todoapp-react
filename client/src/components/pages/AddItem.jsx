import React from "react";

const AddItem = (props) => {
  return (
    <>
      <label> Enter the new todo item</label>
      <input
        type="text"
        value={props.currentTodo}
        onChange={props.updateCurrentTodo}
      />
      <button onClick={props.addTodo}>ADD</button>
      <br />
      <br />
    </>
  );
};

export default AddItem;
