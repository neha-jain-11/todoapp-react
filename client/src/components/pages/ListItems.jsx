import React from "react";

const ListItems = (props) => {
  return (
    <>
      <ul>
        {props.todos.map((todo, index) => {
          return (
            <>
              <li key={index}>
                <span
                  data-todo-id={todo.id}
                  onClick={props.updateStatus}
                  className={todo.status ? "li-span-strike" : ""}>
                  {todo.name}
                </span>
                &nbsp; &nbsp;&nbsp; &nbsp;&nbsp; &nbsp;
                <button data-id={todo.id} onClick={props.deleteTodo}>
                  DELETE
                </button>
              </li>
            </>
          );
        })}
      </ul>
      <br />
    </>
  );
};

export default ListItems;
