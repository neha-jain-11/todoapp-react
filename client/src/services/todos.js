import axios from "axios";

export const getTodos = async () => {
  const res = await axios.get("/api/todos/get");
  return res.data;
};
