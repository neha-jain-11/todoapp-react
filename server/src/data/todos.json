[
  {
    "name": "one",
    "description": "one"
  },
  {
    "name": "two",
    "description": "two"
  }
]
