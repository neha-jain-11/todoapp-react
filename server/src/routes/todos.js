const express = require("express");
const todos = require("../data/todos.json");
const todoRouter = express.Router();

//middleware router level
todoRouter.use("/get", (req, res, next) => {
  console.log("cool cool");
  next();
});

todoRouter.get("/get", (req, res) => {
  res.json(todos);
  res.status(200);
});

todoRouter.post("/post", (req, res) => {
  console.log("hi saving todos", req.body);
  res.send("Saved");
  res.status(200);
});

module.exports = todoRouter;
