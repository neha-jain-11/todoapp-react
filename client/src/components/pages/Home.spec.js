import React from "react";
import { shallow } from "enzyme";
import Home from "./Home.jsx";
import AddItem from "./AddItem.jsx";
import ListItems from "./ListItems.jsx";
import * as Todos from "../../services/todos";

describe("Home test", () => {
  let wrapper;
  let mockService;

  beforeEach(() => {
    mockService = jest
      .spyOn(Todos, "getTodos")
      .mockImplementation(() => [{ id: "11" }]);
    wrapper = shallow(<Home />);
  });

  it("should be renderend properly", () => {
    expect(wrapper).not.toBe.null;
    expect(wrapper.find(AddItem).length).toEqual(1);
    expect(wrapper.find(AddItem)).toBeTruthy;
    expect(wrapper.find(ListItems).length).toEqual(1);
    expect(wrapper.find("button").length).toEqual(1);
  });

  it("should be renderend Additem properly", (done) => {
    const addItem = wrapper.find(AddItem);
    expect(addItem.props().currentTodo).toEqual("");
    expect(mockService).toHaveBeenCalledTimes(1);

    addItem.props().updateCurrentTodo({ target: { value: "abc" } });
    setTimeout(() => {
      expect(wrapper.state().currentTodo).toEqual("abc");
      done();
    });
  });

  it("snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
