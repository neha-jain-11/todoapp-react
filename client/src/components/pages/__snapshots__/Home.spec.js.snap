// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test snapshot 1`] = `ShallowWrapper {}`;
